conans/client/conan_api.py

```diff
@@ -1249,53 +1272,90 @@ def update_lock(self, old_lockfile, new_lockfile, cwd=None):
+        graph_lock = graph_lock_file.graph_lock
+        build_order = graph_lock.build_order()
         return build_order
 
     @api_method
     def lock_clean_modified(self, lockfile, cwd=None):
         cwd = cwd or os.getcwd()
         lockfile = _make_abs_path(lockfile, cwd)
-        lock = GraphLockFile.load(lockfile, self.app.config.revisions_enabled)
-        lock.graph_lock.clean_modified()
-        lock.save(lockfile)
+
+        graph_lock_file = GraphLockFile.load(lockfile, self.app.cache.config.revisions_enabled)
+        graph_lock = graph_lock_file.graph_lock
+        graph_lock.clean_modified()
+        graph_lock_file.save(lockfile)
 
     @api_method
-    def create_lock(self, reference, remote_name=None, settings=None, options=None, env=None,
-                    profile_names=None, update=False, lockfile=None, build=None, profile_build=None):
-        profile_host = ProfileData(profiles=profile_names, settings=settings, options=options, env=env)
-        reference, graph_info = self._info_args(reference, None, profile_host, profile_build)
+    def lock_create(self, path, reference=None, name=None, version=None, user=None, channel=None,
+                    profile_host=None, profile_build=None, remote_name=None, update=None, build=None,
+                    base=None, lockfile=None, lockfile_out=None):
+        # profile_host is mandatory
+        profile_host = profile_host or ProfileData(None, None, None, None)
+        cwd = get_cwd()
+
+        if path and reference:
+            raise ConanException("Both path and reference were provided")
+
+        if path:
+            ref_or_path = _make_abs_path(path, cwd)
+            if not os.path.isfile(ref_or_path):
+                raise ConanException("Conanfile does not exist in %s" % ref_or_path)
+        else:  # reference
+            ref_or_path = ConanFileReference.loads(reference)
+
+        phost = pbuild = graph_lock = None
+        if lockfile:
+            lockfile = _make_abs_path(lockfile, cwd)
+            graph_lock_file = GraphLockFile.load(lockfile, self.app.cache.config.revisions_enabled)
+            phost = graph_lock_file.profile_host
+            pbuild = graph_lock_file.profile_build
+            graph_lock = graph_lock_file.graph_lock
+            graph_lock.relax()
+
+        if not phost:
+            phost = profile_from_args(profile_host.profiles, profile_host.settings,
+                                      profile_host.options, profile_host.env, cwd, self.app.cache)
+
+        if profile_build and not pbuild:
+            # Only work on the profile_build if something is provided
+            pbuild = profile_from_args(profile_build.profiles, profile_build.settings,
+                                       profile_build.options, profile_build.env, cwd, self.app.cache)
+
+        root_ref = ConanFileReference(name, version, user, channel, validate=False)
```

Comment:
```suggestion
        root_ref = ConanFileReference(name, version, user, channel, validate=True)
```

These are the raw arguments from the command line? We can validate them now.